src/test/java/com/microsoft/jfr/RecordingOptionsTest.java

```diff
@@ -0,0 +1,272 @@
+package com.microsoft.jfr;
+
+import org.testng.annotations.DataProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.*;
+
+public class RecordingOptionsTest {
+
+    @DataProvider(name="nameValues")
+    public static Object[][] nameValues() {
+        return new Object[][] {
+                {"test", "test"},
+                {" test", "test"},
+                {" test ", "test"},
+                {"", ""},
+                {null, ""}
+        };
+    }
+    @org.testng.annotations.Test(dataProvider = "nameValues")
+    public void testGetName(String[] args) {
```

Comment:
Prefer test method names to refer to the assert - what is it you're really testing?